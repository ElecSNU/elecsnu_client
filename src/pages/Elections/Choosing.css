#choosing {
    height: 100%;
    padding: 5%;
    width: 100%;
    display: flex;
    flex-direction: column;
    justify-content: space-between;
    align-items: center;
}

#choosing > * {
    width: 100%;
}

.candidate {
    padding: 2.5%;
    margin: 2.5%;
    transition: all 0.7s;
    border: 5px solid transparent;
    cursor: pointer;
}

.candidate h1 {
    transition: all 0.7s;
}

.candidate:hover,
.candidate.chosen {
    border: 5px solid var(--accent1);
}

.candidate:hover h1,
.candidate.chosen h1 {
    color: var(--accent1);
}

#choosing .submit-button {
    width: 50%;
}
