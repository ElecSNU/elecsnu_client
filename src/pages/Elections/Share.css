#share {
    width: 100vw;
    height: 100vh;
    position: absolute;
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%);
    background-color: rgba(0, 0, 0, 0.7);
    display: grid;
    place-items: center;
    animation: show 0.7s ease-in-out;
}

@keyframes show {
    0% {
        opacity: 0;
    }

    100% {
        opacity: 1;
    }
}

.share-modal {
    position: relative;
    width: 75%;
}

.cross {
    font-size: 2rem;
    color: red;
    position: absolute;
    top: 10%;
    right: 2.5%;
    cursor: pointer;
}
