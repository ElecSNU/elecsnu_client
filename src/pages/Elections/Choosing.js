import React, { useState } from 'react';
import useFirestore from '../../hooks/useFirestore';
import { fire_store } from '../../configs/firebase/config';
import { toast } from 'react-toastify';
import useLoader from '../../hooks/useLoader';

import Share from './Share';

import './Choosing.css';

const Choosing = ({ candidates }) => {
    const [chosen, setChosen] = useState(0);
    const electionDocs = useFirestore('newelections').docs;
    const showLoader = useLoader();
    const [done, setDone] = useState(false);
    const [electionName, setElectionName] = useState('');

    const castVote = () => {
        showLoader(true);
        const electionId = window.location.pathname.split(
            '/'
        )[3];
        let electionDetails = {};
        electionDocs.forEach((e) => {
            if (e.id === electionId) {
                electionDetails = e;
                return;
            }
        });
        setElectionName(electionDetails.election_name);

        let userGender = window.localStorage.getItem(
            'voter_gender'
        );
        let userDept = window.localStorage.getItem(
            'voter_dept'
        );
        let userBatch = window.localStorage.getItem(
            'voter_batch'
        );
        let userRoll = window.localStorage.getItem(
            'user_roll'
        );

        electionDetails.voters = [
            ...electionDetails.voters,
            Number(userRoll),
        ];
        electionDetails.election_votes = [
            ...electionDetails.election_votes,
            {
                voter_gender: userGender,
                groups: `${userDept},Batch${userBatch}`,
                candidate_chosen: chosen,
            },
        ];

        fire_store
            .collection('newelections')
            .doc(electionId)
            .set(electionDetails);

        showLoader(false);

        toast.success('Successfully casted vote!');

        setDone(true);
    };

    return (
        <>
            {done ? (
                <Share electionName={electionName} />
            ) : (
                <></>
            )}
            <div id='choosing'>
                <div className='heading-text capital-text center-text foreground-back'>
                    Who would you want to vote for?
                </div>
                <div className='candidates'>
                    {candidates.map((candidate, index) => {
                        return (
                            <div
                                className={`candidate background-1 border-radius-15 ${
                                    index === chosen
                                        ? 'chosen'
                                        : ''
                                }`}
                                key={index}
                                onClick={() =>
                                    setChosen(index)
                                }
                            >
                                <h1 className='capital-text subheading-text foreground-light center-text'>
                                    {candidate}
                                </h1>
                            </div>
                        );
                    })}
                </div>
                <button
                    className='submit-button'
                    onClick={castVote}
                >
                    CAST VOTE
                </button>
            </div>
        </>
    );
};

export default Choosing;
