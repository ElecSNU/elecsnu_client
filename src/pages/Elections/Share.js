import React from 'react';

import './Share.css';

// import icons
import FaTwitter from 'react-icons/lib/fa/twitter';
import FaFacebook from 'react-icons/lib/fa/facebook';
import FaEnvelope from 'react-icons/lib/fa/envelope';
import FaLinkedin from 'react-icons/lib/fa/linkedin';

// import react-custom-share components
import {
    ShareButtonRectangle,
    ShareBlockStandard,
} from 'react-custom-share';

const ShareComponent = (props) => {
    // create object with props for shareBlock
    const shareBlockProps = {
        url: 'elecsnu.firebaseapp.com',
        button: ShareButtonRectangle,
        buttons: [
            { network: 'Twitter', icon: FaTwitter },
            { network: 'Facebook', icon: FaFacebook },
            { network: 'Email', icon: FaEnvelope },
            { network: 'Linkedin', icon: FaLinkedin },
        ],
        text: 'Voting using ElecSNU',
        longtext: `Hey! I just voted for ${props.name} using ElecSNU. Check out their website for more!`,
    };

    return <ShareBlockStandard {...shareBlockProps} />;
};

const Share = ({ electionName }) => {
    return (
        <div id='share'>
            <div className='share-modal'>
                <h1
                    className='heading-text center-text foreground-accent1 border-radius-15 background-1'
                    style={{
                        padding: '2.5%',
                    }}
                >
                    Share your voting experience with
                    everyone!
                </h1>
                <div
                    className='cross'
                    onClick={() => {
                        window.location.pathname = '/';
                    }}
                >
                    x
                </div>
                <ShareComponent name={electionName} />
            </div>
        </div>
    );
};

export default Share;
