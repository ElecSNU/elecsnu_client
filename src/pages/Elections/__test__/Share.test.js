import { render, screen, fireEvent, cleanup } from '@testing-library/react';
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import Choosing from './../Share'
import { toast } from 'react-toastify';
import React, { useEffect, useState } from 'react';
import Share from './../Share';
// import Elections from './../Elections';
// import ParticleBackground from '../../components/ParticleBackground/ParticleBackground';


test('Tests the Sharing ability', () => {
    const div = document.createElement("div")
    const tree = renderer.create(<Share electionName={'class cr'}></Share>).toJSON();
    expect(tree).toMatchSnapshot(); 
   });