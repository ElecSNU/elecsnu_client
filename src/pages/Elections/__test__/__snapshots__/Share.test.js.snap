// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests the Sharing ability 1`] = `
<div
  id="share"
>
  <div
    className="share-modal"
  >
    <h1
      className="heading-text center-text foreground-accent1 border-radius-15 background-1"
      style={
        Object {
          "padding": "2.5%",
        }
      }
    >
      Share your voting experience with everyone!
    </h1>
    <div
      className="cross"
      onClick={[Function]}
    >
      x
    </div>
    <div
      className="css-1qq1qhc"
    >
      
      <div
        className=" rcs-buttons"
      >
        <button
          aria-label="Share on Twitter"
          className="css-trk4o9 rcs-button"
          onClick={[Function]}
        >
          <svg
            fill="currentColor"
            height="1em"
            preserveAspectRatio="xMidYMid meet"
            style={
              Object {
                "color": undefined,
                "verticalAlign": "middle",
              }
            }
            viewBox="0 0 40 40"
            width="1em"
          >
            <g>
              <path
                d="m37.7 9.1q-1.5 2.2-3.7 3.7 0.1 0.3 0.1 1 0 2.9-0.9 5.8t-2.6 5.5-4.1 4.7-5.7 3.3-7.2 1.2q-6.1 0-11.1-3.3 0.8 0.1 1.7 0.1 5 0 9-3-2.4-0.1-4.2-1.5t-2.6-3.5q0.8 0.1 1.4 0.1 1 0 1.9-0.3-2.5-0.5-4.1-2.5t-1.7-4.6v0q1.5 0.8 3.3 0.9-1.5-1-2.4-2.6t-0.8-3.4q0-2 0.9-3.7 2.7 3.4 6.6 5.4t8.3 2.2q-0.2-0.9-0.2-1.7 0-3 2.1-5.1t5.1-2.1q3.2 0 5.3 2.3 2.4-0.5 4.6-1.7-0.8 2.5-3.2 3.9 2.1-0.2 4.2-1.1z"
              />
            </g>
          </svg>
        </button>
        <button
          aria-label="Share on Facebook"
          className="css-hn5fd rcs-button"
          onClick={[Function]}
        >
          <svg
            fill="currentColor"
            height="1em"
            preserveAspectRatio="xMidYMid meet"
            style={
              Object {
                "color": undefined,
                "verticalAlign": "middle",
              }
            }
            viewBox="0 0 40 40"
            width="1em"
          >
            <g>
              <path
                d="m29.4 0.3v5.9h-3.5q-1.9 0-2.6 0.8t-0.7 2.4v4.2h6.6l-0.9 6.6h-5.7v16.9h-6.8v-16.9h-5.7v-6.6h5.7v-4.9q0-4.1 2.3-6.4t6.2-2.3q3.3 0 5.1 0.3z"
              />
            </g>
          </svg>
        </button>
        <button
          aria-label="Share on Email"
          className="css-1pclnts rcs-button"
          onClick={[Function]}
        >
          <svg
            fill="currentColor"
            height="1em"
            preserveAspectRatio="xMidYMid meet"
            style={
              Object {
                "color": undefined,
                "verticalAlign": "middle",
              }
            }
            viewBox="0 0 40 40"
            width="1em"
          >
            <g>
              <path
                d="m40 15.8v17.8q0 1.4-1 2.5t-2.6 1h-32.8q-1.5 0-2.6-1t-1-2.5v-17.8q1 1.1 2.3 2 8 5.5 11 7.7 1.3 0.9 2.1 1.5t2.1 1 2.5 0.6h0q1.2 0 2.5-0.6t2.1-1 2.1-1.5q3.7-2.8 11.1-7.7 1.2-0.9 2.2-1.9z m0-6.5q0 1.7-1.1 3.4t-2.7 2.7q-8.4 5.8-10.5 7.3-0.2 0.1-0.9 0.6t-1.2 0.9-1.2 0.7-1.3 0.6-1.1 0.2h0q-0.5 0-1.1-0.2t-1.3-0.6-1.2-0.7-1.2-0.9-0.9-0.6q-2.1-1.5-5.9-4.1t-4.6-3.2q-1.3-0.9-2.6-2.6t-1.2-3q0-1.8 0.9-2.9t2.7-1.2h32.8q1.5 0 2.5 1.1t1.1 2.5z"
              />
            </g>
          </svg>
        </button>
        <button
          aria-label="Share on Linkedin"
          className="css-1pjeie1 rcs-button"
          onClick={[Function]}
        >
          <svg
            fill="currentColor"
            height="1em"
            preserveAspectRatio="xMidYMid meet"
            style={
              Object {
                "color": undefined,
                "verticalAlign": "middle",
              }
            }
            viewBox="0 0 40 40"
            width="1em"
          >
            <g>
              <path
                d="m10.8 13.9v22.2h-7.4v-22.1h7.4z m0.5-6.8q0 1.7-1.2 2.7t-3 1.1h0q-1.9 0-3-1.1t-1.1-2.7q0-1.6 1.2-2.7t3-1.1 2.9 1.1 1.2 2.7z m26 16.3v12.7h-7.4v-11.9q0-2.3-0.9-3.6t-2.8-1.4q-1.4 0-2.3 0.8t-1.5 1.9q-0.2 0.7-0.2 1.8v12.4h-7.3q0-8.9 0-14.5t0-6.6l0-1h7.3v3.2h-0.1q0.5-0.7 1-1.3t1.2-1.1 2-1 2.5-0.4q3.8 0 6.2 2.6t2.3 7.4z"
              />
            </g>
          </svg>
        </button>
      </div>
    </div>
  </div>
</div>
`;
